src/algebra/algebra/spectrum.lean

```diff
@@ -0,0 +1,187 @@
+/-
+Copyright (c) 2021 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import algebra.algebra.basic
+import tactic.noncomm_ring
+/-!
+# Spectrum of an element in an algebra
+This file develops the basic theory of the spectrum of an element of an algebra.
+This theory will serve as the foundation for spectral theory in Banach algebras.
+
+## Main definitions
+
+* `resolvent a`: has type `set R` is the resolvent set of an element `a : A` where
+  `A` is an  `R`-algebra.
+* `spectrum a`: has type `set R` is the spectrum of an element `a : A` where
+  `A` is an  `R`-algebra.
+
+## Main statements
+
+* `unit_left_coset_spectrum`: units in the scalar ring commute (multplication) with the spectrum.
+* `left_add_coset_spectrum`: elements of the scalar ring commute (addition) with the spectrum.
+* `unit_mem_spectrum_mul_iff_swap_mul`: the units (of `R`) in `σ (a*b)` coincide
+  with those in `σ (b*a)`.
+
+## Notations
+
+* `σ a` : `spectrum R a` of `a : A`
+
+## TODO
+
+* Prove the *spectral mapping theorem* for the polynomial functional calculus
+-/
+
+universes u v
+
+namespace spectrum
+
+section defs
+
+variables (R : Type u) {A : Type v}
+variables [comm_ring R] [ring A] [algebra R A]
+
+-- definition and basic properties
+
+/-- Given a commutative ring `R` and an `R`-algebra `A`, the *resolvent* of `a : A`
+is the `set R` consisting of those `r : R` for which `r•1 - a` is a unit of the
+algebra `A`.  -/
+definition resolvent (a : A) : set R :=
+{ r : R | is_unit (r • 1 - a) }
```

Comment:
I think we usually spell this
```suggestion
{ r : R | is_unit (algebra_map R A r - a) }
```

Reply:
Is this something I should change throughout, or just here? It seems a bit cumbersome to have everywhere.

Reply:
Yes, it would be throughout

Reply:
Perhaps wait for the opinion of a second reviewer

Reply:
To me `r • 1` looks a lot nicer, so if there's no big downside (like the API being more developed for that version or something like that) I would vote for that.

Reply:
I made the suggested change. With the notation Eric suggested in Zulip, it really wasn't too bad.